src/mono/mono/mini/interp/transform-simd.c

```diff
@@ -424,7 +429,13 @@ emit_sri_vector128 (TransformData *td, MonoMethod *cmethod, MonoMethodSignature
 	gint16 simd_opcode = -1;
 	gint16 simd_intrins = -1;
 
-	vector_klass = mono_class_from_mono_type_internal (csignature->ret);
+	if (csignature->ret->type == MONO_TYPE_GENERICINST) {
```

Comment:
It'd be cool to see a comment listing which methods (if any) this doesn't cover. I glanced over the list of Vector128 methods and didn't notice any obvious ones, but I figure you would know for sure. It feels a little "magical".

Reply:
It's mostly just a future proofing check, we have a similar thing in RyuJIT.

The runtime needs to generally be robust in the face of arbitrary corelib implementations or future revisions and it's always possible we could have an API like `T M(T, Vector128<T> x)` for example, in which case we want to ensure we fallback to managed rather than letting the runtime crash.